### PR TITLE
Ssh users

### PR DESCRIPTION
**Pull request by whites11**
Towards: https://github.com/giantswarm/giantswarm/issues/13783

This PR uses the list of ssh public keys coming from draughtsman (as an helm parameter in the configmap) to generate the ignition config file, ignoring the field in AzureConfig CR.

We need to do this otherwise the customer could inject their SSH keys in tenant clusters.

### CODE REVIEW

File: service/controller/azure_machine_pool.go
Comment (fiunchinho) [inline service/controller/azure_machine_pool.go:58]: Can't we keep the type to `string`? I think there is value in having `azure_machine_pool` controller not depending on `v1alpha1` types, and only use go types or CAPI/CAPZ types.

Comment (whites11) [reply service/controller/azure_machine_pool.go:58]: we can, at the cost of duplicating the parsing function in two places (or moving it into a shared location).
What about having a new local type to hold this information?

Comment (fiunchinho) [reply service/controller/azure_machine_pool.go:58]: Duplication is fine sometimes, specially having in mind that `AzureConfig` related stuff will be removed in the following months. Anyway, your call.

Comment (whites11) [reply service/controller/azure_machine_pool.go]: I created an external module for parsing the flag and pass around a new custom type between controllers.
WDYT?

**Review by fiunchinho — APPROVED** [review]
Awesome!

**Pull request merged**
